src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/AbstractFileWriter.cs

```diff
@@ -70,7 +70,11 @@ protected void WriteLine()
 
         protected void WriteLine(string msg)
         {
-            WriteIndentIfNeeded();
+            if (msg != "")
+            {
+                WriteIndentIfNeeded();
+            }
```

Comment:
Prevents people from calling `WriteLine("")` and having that actually emit pointless whitespace.

Reply:
Since that file seems to be in a proprietary part, I couldn't check the context, but provided `msg` is a string, would perhaps `string.IsNullOrEmpty` be more prudent?

Reply:
> Since that file seems to be in a proprietary part, 

I'm not sure what that means @tamlin-mike

> would perhaps string.IsNullOrEmpty be more prudent?

Callers aren't passing in null here :)

Reply:
> I'm not sure what that means

@CyrusNajmabadi Clicking the file link resulted in "We went looking everywhere, but couldn’t find those commits". Searching for that filename in the roslyn repo didn't show anything. That suggested to me it's in a proprietary non-public part of the roslyn repository only Microsoft and its apointees have access to.

> Callers aren't passing in null here :)

If that can be assured with 100% certainty, now and forever, OK.

Reply:
Maybe make it a `Debug.Assert( text != null )` ?

Reply:
> That suggested to me it's in a proprietary non-public part of the roslyn repository only Microsoft and its apointees have access to.

I'm not MS, or an appointee of it :-)

The file can't be found due to quirks around how GitHub works with force-pushes.  If you just view the pr changes, you'll see this change

Reply:
@adamspeight2008 I don't really see any value there. Thanks